### DISCUSSION TITLE
Integrazione ESPHome

### DISCUSSION THREAD

**Discussion by serucci** [indomus/forum#728] [HUB: Home Assistant]
Ciao a tutti, perdonate la banalità della domanda, in homeassistant vorrei integrare i dispositivi esphome, nel campo host mi chiede ,Inserisci le impostazioni di connessione del tuo nodo ESPHome...a cosa si riferisce?

**Comment by snakuzzo** [comment]
All'indirizzo ip.
Comunque dovrebbe trovarli in automatico (a patto che tu abbia attivo il discovery)

**Reply by serucci** [reply]
Scusami non capisco, calcola che sto iniziando a muovermi in questo campo adesso....mi spiego : vorrei installare il firmware di un progetto per la gestione di una centrale antifurto tramite rs232 su un dispostivo ESP8266 ,gestito poi da Home Assitant . ho installato su pc ,win10, home assistant e adesso dovrei  fare l'integrazione di ESPHome.  Per indirizzo ip intendi quello della mia schedina esp ? Come si attiva il discovery?

**Reply by indomus** [reply]
https://indomus.it/guide/integrare-componenti-esphome-a-home-assistant-via-api/

**Reply by snakuzzo** [reply]
Innanzi tutto devi distinguere l'addon esphome dall'integrazione.

Che tipo di installazione hai fatto?
VM HassOS, Core in venv o Core su Docker ?

Se hai una VM HassOS dovresti scaricare l'addon esphome che ti mette anche a disposizione la dashboard (opzionale di base) per poter gestire i tuoi device esphome, incluso il flash dei device

Se hai installazione Core invece, non hai a disposizione alcun addon...e devi procedere con installazione di esphome (è semplicemente un pip install) oppure scaricando l'immagine Docker di esphome e usare quella.

Poi...
una volta flashati i device, questi si agganceranno alla tua WIFI e... su HA, andando su Integrazioni dovresti trovare già discoverati i tuoi device esphome a patto che, come ti dicevo nel mio precedente post, sia attivo il discovery su HA...altrimenti li aggiungerai semplicemente indicando l'IP LAN del device.

Il discovery è attivo di default se usi defalut config, altrimenti per avere il discovery di esphome devi attivare zeroconf
Leggi la [documentazione](https://www.home-assistant.io/integrations/discovery/)

Spero di essere stato chiaro

**Reply by serucci** [reply]
Ho fatto una installazione core. Bene farò l'installazione dell'addon esphome tramite pip.
Il device si flasha come?

**Reply by indomus** [reply]
Ciao serucci,
non è cattiveria, ma passiamo settimane a scrivere guide apposta per non dover spiegare 1:1 le cose sul forum.

Trovi tutte le risposte alle tue domande, [qui](https://indomus.it/componenti/esphome/).
